presto-main/src/test/java/com/facebook/presto/operator/TestDynamicFilterSourceOperator.java

```diff
@@ -0,0 +1,300 @@
+                          new Page(createBooleansBlock(false, 1), createDoublesBlock(4.5)));
+        operatorFactory.noMoreOperators();
+
+        assertEquals(partitions.build(), ImmutableList.of(
+                TupleDomain.withColumnDomains(ImmutableMap.of(
+                        "1", Domain.multipleValues(DOUBLE, ImmutableList.of(1.5, 3.0, 4.5))))));
+    }
+
+    @Test
+    public void testCollectWithNulls()
+    {
+        Block blockWithNulls = INTEGER
+                .createFixedSizeBlockBuilder(0)
+                .writeInt(3)
+                .appendNull()
+                .writeInt(4)
+                .build();
+
+        OperatorFactory operatorFactory = createOperatorFactory(channel(0, INTEGER));
+        verifyPassthrough(createOperator(operatorFactory),
+                          ImmutableList.of(INTEGER),
+                          new Page(createLongsBlock(1, 2, 3)),
+                          new Page(blockWithNulls),
+                          new Page(createLongsBlock(4, 5)));
+        operatorFactory.noMoreOperators();
+
+        assertEquals(partitions.build(), ImmutableList.of(
+                TupleDomain.withColumnDomains(ImmutableMap.of(
+                        "0", Domain.create(ValueSet.of(INTEGER, 1L, 2L, 3L, 4L, 5L), false)))));
+    }
+
+    @Test
+    public void testCollectNoFilters()
+    {
+        OperatorFactory operatorFactory = createOperatorFactory();
+        verifyPassthrough(createOperator(operatorFactory),
+                          ImmutableList.of(BIGINT),
+                          new Page(createLongsBlock(1, 2, 3)));
+        operatorFactory.noMoreOperators();
+        assertEquals(partitions.build(), ImmutableList.of(TupleDomain.all()));
+    }
+
+    @Test
+    public void testCollectEmptyBuildSide()
+    {
+        OperatorFactory operatorFactory = createOperatorFactory(channel(0, BIGINT));
+        verifyPassthrough(createOperator(operatorFactory),
+                          ImmutableList.of(BIGINT));
+        operatorFactory.noMoreOperators();
+        assertEquals(partitions.build(), ImmutableList.of(TupleDomain.none()));
+    }
+
+    @Test
+    public void testCollectTooMuchRows()
+    {
+        final int maxRowCount = getDynamicFilteringMaxPerDriverRowCount(pipelineContext.getSession());
+        Page largePage = createSequencePage(ImmutableList.of(BIGINT), maxRowCount + 1);
+
+        OperatorFactory operatorFactory = createOperatorFactory(channel(0, BIGINT));
+        verifyPassthrough(createOperator(operatorFactory),
+                          ImmutableList.of(BIGINT),
+                          largePage);
+        operatorFactory.noMoreOperators();
+        assertEquals(partitions.build(), ImmutableList.of(TupleDomain.all()));
+    }
+
+    @Test
+    public void testCollectTooMuchBytesSingleColumn()
+    {
+        final long maxByteSize = getDynamicFilteringMaxPerDriverSize(pipelineContext.getSession()).toBytes();
```

Comment:
This class has many conflicts with our coding style. Please verify each of them.